### PR TITLE
Fix to essence cost multiplied by votes

### PR DESCRIPTION
**Pull request by lily-pytel**
It's basically the same as when it was being multiplied by factions. Joins... ugh. The view needs to be regenerated to be fixed so hopefully the migration takes care of it.

### CODE REVIEW

File: postgres/init.sql [hunk 1]
```diff
@@ -380,7 +380,22 @@ RETURNS INTEGER AS $$
...
+    SELECT count(DISTINCT deck_vote)::int into votes
+    FROM mythgard.deck
+   LEFT JOIN mythgard.deck_vote
+     On deck_vote.deck_id = deck.id
+   WHERE deck.id = $1
+   GROUP BY deck.id;
+    
+    RETURN votes;
   END;
   $$ language 'plpgsql';
```
Comment (alexmiddeleer) [inline postgres/init.sql]: Probably could just be `language sql`, and then you don't need the SELECT into stuff. See other `language sql` funcs.

File: postgres/init.sql [hunk 2]
```diff
@@ -400,10 +415,6 @@ CREATE OR REPLACE VIEW mythgard.deck_preview as
     ON (card.id = card_faction.card_id and card_faction.faction_id is not null)
   LEFT JOIN mythgard.faction
     On faction.id = card_faction.faction_id
-  LEFT JOIN mythgard.essence_costs
-    On essence_costs.rarity = card.rarity
```
Comment (alexmiddeleer) [inline postgres/init.sql]: Oops, I guess that join was basically dead code

File: postgres/init.sql [hunk 3]
```diff
@@ -390,7 +405,7 @@ CREATE OR REPLACE VIEW mythgard.deck_preview as
          deck.created as deck_created,
          array_agg(DISTINCT faction.name) as factions,
          deck_essence_cost(deck.id)::int as essence_cost,
```
Comment (alexmiddeleer) [inline postgres/init.sql]: Maybe we just needed `deck_essence_cost(distinct deck.id)::int as essence_cost,`? If that's even legal?

File: postgres/init.sql [hunk 4]
```diff
@@ -380,7 +380,22 @@ RETURNS INTEGER AS $$
     GROUP BY deck.id;
     
     RETURN essence_cost;
-  
+  END;
+  $$ language 'plpgsql';
+
+CREATE OR REPLACE FUNCTION deck_votes (IN deckId INTEGER)
```
Comment (alexmiddeleer) [inline postgres/init.sql]: should be mythgard.deck_votes probably

File: postgres/init.sql [hunk 5]
```diff
@@ -390,7 +405,7 @@ CREATE OR REPLACE VIEW mythgard.deck_preview as
          deck.created as deck_created,
          array_agg(DISTINCT faction.name) as factions,
          deck_essence_cost(deck.id)::int as essence_cost,
-         count(DISTINCT deck_vote)::int as votes
+         deck_votes(deck.id)::int as votes
```
Comment (alexmiddeleer) [inline postgres/init.sql]: mythgard.deck_votes

File: postgres/init.sql (hunk 1 above)
Comment (lily-pytel) [reply postgres/init.sql]: Why did we do `update_modified_column` in `plpgsql`? Also I kind of like it more - looks more like a normal function to me

File: postgres/init.sql (hunk 3 above)
Comment (lily-pytel) [reply postgres/init.sql]: Nooo it did not like that. I'm not that sure what that was supposed to do but the compiler or builder or whatever was not a fan

File: postgres/init.sql (hunk 1 above)
Comment (alexmiddeleer) [reply postgres/init.sql]: @lily-pytel That's cool. The answer to the first question is probably that I copy pasted it from something.

File: postgres/init.sql (hunk 3 above)
Comment (alexmiddeleer) [reply postgres/init.sql]: /shrug okies

**Pull request merged**